pkg/virt-launcher/virtwrap/network/podinterface_test.go

```diff
@@ -587,6 +587,92 @@ var _ = Describe("Pod Network", func() {
+					mockNetwork.EXPECT().LinkByName(primaryPodInterfaceName).Return(primaryPodInterface, nil)
+
+					srcAddressesToSnat := []string{getLoopbackAdrress(proto)}
+					dstAddressesToDnat := []string{getLoopbackAdrress(proto)}
+					if proto == iptables.ProtocolIPv4 {
+						srcAddressesToSnat = append(srcAddressesToSnat, getEnvoyLoopbackAddress())
+						dstAddressesToDnat = append(dstAddressesToDnat, fakeAddr.IP.String())
+					}
+					mockNetwork.EXPECT().NftablesAppendRule(proto, "nat",
+						"KUBEVIRT_POSTINBOUND", GetNFTIPString(proto), "saddr", fmt.Sprintf("{ %s }", strings.Join(srcAddressesToSnat, ", ")),
+						"counter", "snat", "to", GetMasqueradeGwIp(proto)).Return(nil)
+					mockNetwork.EXPECT().NftablesAppendRule(proto, "nat",
+						"output", GetNFTIPString(proto), "daddr", fmt.Sprintf("{ %s }", strings.Join(dstAddressesToDnat, ", ")),
+						"counter", "dnat", "to", GetMasqueradeVmIp(proto)).Return(nil)
+				}
+
+				domain := NewDomainWithBridgeInterface()
+				vm := newVMIMasqueradeInterface("testnamespace", "testVmName")
+				vm.Annotations = map[string]string{
+					IstioInjectAnnotation: "true",
+				}
+
+				api.NewDefaulter(runtime.GOARCH).SetObjectDefaults_Domain(domain)
+				TestPodInterfaceIPBinding(vm, domain)
+			})
+			It("should define a new DhcpConfig bind to a bridge with Istio enabled and create a specific nat rule using nftables", func() {
+				// Forward a specific port
+				mockNetwork.EXPECT().IsIpv6Enabled(primaryPodInterfaceName).Return(true, nil).Times(3)
+				mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil).Times(1)
+
+				for _, proto := range [2]iptables.Protocol{iptables.ProtocolIPv4, iptables.ProtocolIPv6} {
+					mockNetwork.EXPECT().NftablesLoad(proto).Return(nil)
+
+					mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return(fakeAddr.IP.String(), "", nil)
+					mockNetwork.EXPECT().LinkByName(primaryPodInterfaceName).Return(primaryPodInterface, nil)
+
+					srcAddressesToSnat := []string{getLoopbackAdrress(proto)}
+					dstAddressesToDnat := []string{getLoopbackAdrress(proto)}
+					if proto == iptables.ProtocolIPv4 {
+						srcAddressesToSnat = append(srcAddressesToSnat, getEnvoyLoopbackAddress())
+						dstAddressesToDnat = append(dstAddressesToDnat, fakeAddr.IP.String())
+					}
+					mockNetwork.EXPECT().NftablesAppendRule(proto, "nat",
+						"KUBEVIRT_POSTINBOUND",
+						"tcp", "dport", "80",
+						GetNFTIPString(proto), "saddr", fmt.Sprintf("{ %s }", strings.Join(srcAddressesToSnat, ", ")),
+						"counter", "snat", "to", GetMasqueradeGwIp(proto)).Return(nil)
+					mockNetwork.EXPECT().NftablesAppendRule(proto, "nat",
+						"output",
+						GetNFTIPString(proto), "daddr", fmt.Sprintf("{ %s }", strings.Join(dstAddressesToDnat, ", ")),
+						"tcp", "dport", "80",
+						"counter", "dnat", "to", GetMasqueradeVmIp(proto)).Return(nil)
+
+					mockNetwork.EXPECT().NftablesAppendRule(proto, "nat",
```

Comment:
The last two rules seem like a left over.

Reply:
Totally, thank you